Add section in README.md for guidance on installing pygraphviz

I had some trouble installing pygraphviz when trying to run 

```bash
>>> uv venv
>>> source .venv/bin/activate
>>> uv sync --activate --extra nb
```

on MacOS (Apple Silicon M2 Pro). I found the following workaround using 

```bash
brew install graphviz
uv pip install \
  --config-settings=--global-option=build_ext \
  --config-settings=--global-option="-I$(brew --prefix graphviz)/include" \
  --config-settings=--global-option="-L$(brew --prefix graphviz)/lib" \
  pygraphviz==1.14
````

But this should be documented somewhere explicitly or otherwise incorporated into how we install the dependency, perhaps in the README.md

UPDATE: I also ran into this issue when trying to install dependencies from scratch on Ubuntu 24.04 LTS. I ran 

```bash
uv sync --group test --extra gpu
```

and got this error


```
Updated https://github.com/conorheins/pybefit.git (f20c94d30fbf938bf671ee863a552
Resolved 180 packages in 11.66s
warning: The package `jaxlib==0.8.1` does not have an extra named `cuda12`
warning: The package `jaxlib==0.6.2` does not have an extra named `cuda12`
      Built inferactively-pymdp @ file:///home/conor/Documents/.../
      Built pybefit @ git+https://github.com/conorheins/pybefit.git@f20c94d30fbf938b
  × Failed to build `pygraphviz==1.14`
  ├─▶ The build backend returned an error
  ╰─▶ Call to `setuptools.build_meta.build_wheel` failed (exit status: 1)

      [stdout]
      running bdist_wheel
      running build
      running build_py
      creating build/lib.linux-x86_64-cpython-312/pygraphviz
      copying pygraphviz/testing.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz
      copying pygraphviz/graphviz.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz
      copying pygraphviz/agraph.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz
      copying pygraphviz/scraper.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz
      copying pygraphviz/__init__.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz
      creating build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_string.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_clear.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_repr_mimebundle.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_attribute_defaults.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_scraper.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_html.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_unicode.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_node_attributes.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_drawing.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_graph.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_readwrite.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_layout.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_edge_attributes.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_close.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/test_subgraph.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      copying pygraphviz/tests/__init__.py ->
      build/lib.linux-x86_64-cpython-312/pygraphviz/tests
      running egg_info
      writing pygraphviz.egg-info/PKG-INFO
      writing dependency_links to pygraphviz.egg-info/dependency_links.txt
      writing top-level names to pygraphviz.egg-info/top_level.txt
      reading manifest file 'pygraphviz.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'LICENSE'
      writing manifest file 'pygraphviz.egg-info/SOURCES.txt'
      copying pygraphviz/graphviz.i ->
      build/lib.linux-x86_64-cpython-312/pygraphviz
      copying pygraphviz/graphviz_wrap.c ->
      build/lib.linux-x86_64-cpython-312/pygraphviz
      running build_ext
      building 'pygraphviz._graphviz' extension
      creating build/temp.linux-x86_64-cpython-312/pygraphviz
      x86_64-linux-gnu-gcc -fno-strict-overflow -Wsign-compare
      -DNDEBUG -g -O2 -Wall -fPIC -DSWIG_PYTHON_STRICT_BYTE_CHAR
      -I/home/conor/.cache/uv/builds-v0/.tmpYelZTX/include
      -I/usr/include/python3.12 -c pygraphviz/graphviz_wrap.c -o
      build/temp.linux-x86_64-cpython-312/pygraphviz/graphviz_wrap.o

      [stderr]
      /home/conor/.cache/uv/builds-v0/.tmpYelZTX/lib/python3.12/site-packages/setuptools/config/_apply_pyprojecttoml.py:82:
      SetuptoolsDeprecationWarning: `project.license` as a TOML table is
      deprecated
      !!

      
      ********************************************************************************
              Please use a simple string containing a SPDX expression for
      `project.license`. You can also use `project.license-files`. (Both
      options available on setuptools>=77.0.0).

              By 2026-Feb-18, you need to update your project and remove
      deprecated calls
              or your builds will no longer be supported.

              See
      https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license
      for details.
      
      ********************************************************************************

      !!
        corresp(dist, value, root_dir)
      /home/conor/.cache/uv/builds-v0/.tmpYelZTX/lib/python3.12/site-packages/setuptools/config/_apply_pyprojecttoml.py:61:
      SetuptoolsDeprecationWarning: License classifiers are deprecated.
      !!

      
      ********************************************************************************
              Please consider removing the following classifiers in favor of a
      SPDX license expression:

              License :: OSI Approved :: BSD License

              See
      https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license
      for details.
      
      ********************************************************************************

      !!
        dist._finalize_license_expression()
      /home/conor/.cache/uv/builds-v0/.tmpYelZTX/lib/python3.12/site-packages/setuptools/dist.py:759:
      SetuptoolsDeprecationWarning: License classifiers are deprecated.
      !!

      
      ********************************************************************************
              Please consider removing the following classifiers in favor of a
      SPDX license expression:

              License :: OSI Approved :: BSD License

              See
      https://packaging.python.org/en/latest/guides/writing-pyproject-toml/#license
      for details.
      
      ********************************************************************************

      !!
        self._finalize_license_expression()
      warning: no files found matching '*.swg'
      warning: no files found matching '*.png' under directory 'doc'
      warning: no files found matching '*.html' under directory 'doc'
      warning: no files found matching '*.txt' under directory 'doc'
      warning: no files found matching '*.css' under directory 'doc'
      warning: no previously-included files matching '*~' found anywhere in
      distribution
      warning: no previously-included files matching '*.pyc' found anywhere
      in distribution
      warning: no previously-included files matching '.svn' found anywhere
      in distribution
      no previously-included directories found matching 'doc/build'
      pygraphviz/graphviz_wrap.c:9: warning: "SWIG_PYTHON_STRICT_BYTE_CHAR"
      redefined
          9 | #define SWIG_PYTHON_STRICT_BYTE_CHAR
            |
      <command-line>: note: this is the location of the previous definition
      pygraphviz/graphviz_wrap.c:171:11: fatal error: Python.h: No such file
      or directory
        171 | # include <Python.h>
            |           ^~~~~~~~~~
      compilation terminated.
      error: command '/usr/bin/x86_64-linux-gnu-gcc' failed with exit code 1

      hint: This error likely indicates that you need to install a library
      that provides "Python.h" for `pygraphviz@1.14`
  help: `pygraphviz` (v1.14) was included because `inferactively-pymdp:test`
        (v1.0.0) depends on `pygraphviz`
```

--- a/README.md
+++ b/README.md
@@ -87,6 +87,8 @@ If you prefer `pip`, use:
 pip install inferactively-pymdp
 ```
 
+If `uv sync --group test` or `uv sync --extra nb` fails while building `pygraphviz`, install Graphviz first and then retry. On macOS, install `graphviz` with Homebrew. On Ubuntu/Debian, install `graphviz libgraphviz-dev pkg-config build-essential` plus matching Python development headers if `Python.h` is missing. The full troubleshooting notes live in [`docs-mkdocs/getting-started/installation.md`](docs-mkdocs/getting-started/installation.md).
+
 Once in Python, you can then directly import `pymdp`, its sub-packages, and functions.
 
 ```python

--- a/docs-mkdocs/getting-started/installation.md
+++ b/docs-mkdocs/getting-started/installation.md
@@ -28,6 +28,8 @@ uv sync --group test
 - `--extra gpu`: installs CUDA-enabled JAX packages.
 - `--extra docs`: documentation toolchain (`mkdocs`, `mkdocs-material`, `mkdocstrings`, `mkdocs-jupyter`, `mkdocs-redirects`).
 
+`pygraphviz` is only needed for Graphviz-backed notebook visualizations such as the MCTS graph-world demo, but it is included in the current `test` and `nb` dependency sets. On many machines `uv sync --group test` works without any extra steps; if it fails while building `pygraphviz`, use the troubleshooting notes below and then rerun the sync command.
+
 Common `uv` sync patterns:
 
 ```bash
@@ -46,6 +48,78 @@ uv sync --group test --extra docs
 # GPU workflow
 uv sync --group test --extra gpu
 ```
+
+## `pygraphviz` troubleshooting
+
+If installation fails while building `pygraphviz`, the missing dependency is usually one of:
+
+- Graphviz itself, including the C headers (`graphviz/cgraph.h`)
+- Python development headers for the exact interpreter version in your virtual environment (`Python.h`)
+
+### macOS (Homebrew)
+
+Install Graphviz with Homebrew before syncing dependencies:
+
+```bash
+brew install graphviz
+```
+
+If `pygraphviz` still fails to build, install it explicitly with Homebrew's include and library paths:
+
+```bash
+uv pip install \
+  --config-settings=--global-option=build_ext \
+  --config-settings=--global-option="-I$(brew --prefix graphviz)/include" \
+  --config-settings=--global-option="-L$(brew --prefix graphviz)/lib" \
+  pygraphviz==1.14
+```
+
+Then rerun:
+
+```bash
+uv sync --group test
+```
+
+or, if you only need the notebook/media extras:
+
+```bash
+uv sync --extra nb
+```
+
+### Ubuntu / Debian
+
+Install the Graphviz runtime, Graphviz development headers, `pkg-config`, and the compiler toolchain before syncing dependencies:
+
+```bash
+sudo apt-get update
+sudo apt-get install -y graphviz libgraphviz-dev pkg-config build-essential
+```
+
+Install Python development headers that match the interpreter in your environment:
+
+```bash
+sudo apt-get install -y python3-dev
+```
+
+If you are using a version-specific interpreter that is newer than the system default, install the matching package instead, for example:
+
+```bash
+sudo apt-get install -y python3.12-dev
+```
+
+Then rerun:
+
+```bash
+uv sync --group test
+```
+
+or install the dependency directly inside the active environment:
+
+```bash
+uv pip install pygraphviz==1.14
+```
+
+If the build output mentions `graphviz/cgraph.h`, install or verify the Graphviz development packages above. If it mentions `Python.h`, install or verify the matching Python development headers.
 
 ## Verify docs build
 ```bash

